csrc/device_lower/pass/circular_buffer.cpp

```diff
@@ -23,11 +23,13 @@ namespace {
 // buffer tensor is on smem, in which case it would otherwise require
 // an additional predicate to guard buffer overruns. When it's on
 // gmem, that isn't the case, so it does not need to create an
-// epilogue loop.
+// epilogue loop. For TMA cpAsyncBulk, there is always an epilogue loop.
 bool requireEpilogue(const std::vector<Expr*>& exprs) {
   return std::any_of(exprs.begin(), exprs.end(), [](const Expr* expr) {
-    return expr->input(0)->as<TensorView>()->getMemoryType() ==
-        MemoryType::Shared;
+    return (expr->input(0)->as<TensorView>()->getMemoryType() ==
+            MemoryType::Shared) ||
+        (expr->as<LoadStoreOp>()->opType() ==
```

Comment:
Do we need to check `expr->isA<LoadStoreOp>()` here?

Reply:
Switched to `ir_utils::isCpAsyncBulk`.